### PR TITLE
ceph: allow custom labels to be added to the discover daemonset

### PR DESCRIPTION
**Pull request by galexrt**
Signed-off-by: Alexander Trost <galexrt@googlemail.com>

**Description of your changes:**

This allows custom labels to be added to the discover DaemonSet created by the Rook Ceph operator.

The branch is currently based on #6599 to have the parse string to labels function in.

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [x] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [x] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.
- [x] Pending release notes updated with breaking and/or notable changes, if necessary.
- [x] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [x] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: pkg/operator/discover/discover.go [hunk 1]
```diff
@@ -211,6 +212,14 @@ func (d *Discover) createDiscoverDaemonSet(namespace, discoverImage, securityAcc
 		}
 	}
 
+	podLabels := os.Getenv(discoverDaemonSetPodLabelsEnv)
+	if podLabels != "" {
+		podLabels := k8sutil.ParseStringToLabels(podLabels)
+		// Set the app label even if set by user as otherwise the DaemonSet pod selector breaks
+		podLabels["app"] = discoverDaemonsetName
```
Comment (travisn) [inline pkg/operator/discover/discover.go:219]: Is `app` the only label on the discover pod right now? If not, the others would be lost with this change.

**Review by travisn — CHANGES_REQUESTED** [review]
A couple more things, thanks!
- You can rebase now since #6599 merged
- Please add a comment to the pending release notes for this and #6599

File: pkg/operator/discover/discover.go (hunk 1 above)
Comment (galexrt) [reply pkg/operator/discover/discover.go:223]: Yes, that is the only label.
I have just seen that the discover DaemonSet doesn't have any labels added to it and added that the `app` label is set on it.

**Comment by galexrt** [comment]
@travisn I have implemented your feedback and also added missing labels to the discover DaemonSet object. Let me know if the discover DaemonSet object label "fix" should be moved to its own PR. PTAL

**Review by travisn — APPROVED** [review]

**Comment by galexrt** [comment]
Restarted Jenkins CI, seems like a random failure on `aws_1.19.x`.

**Review by BlaineEXE — APPROVED** [review]
Out of curiosity, what is the use case for adding labels to CSI or discover pods?

**Comment by galexrt** [comment]
@BlaineEXE The use case is for monitoring. Depending on how monitoring is realized, labels are used extensively as with the customer I am with right now. The labels allow for easy grouping and "alerting the right people" based on their alert and recording rules.

**Comment by BlaineEXE** [comment]
Gotcha. Thanks!

**Pull request merged**